Kernel/StdLib.h

```diff
@@ -75,6 +76,23 @@ template<typename T>
     return copy_from_user(dest, src, sizeof(T));
 }
 
+template<>
+[[nodiscard]] inline bool copy_from_user<timespec>(timespec* dest, const timespec* src)
```

Comment:
This feels quite weird. I wouldn't expect a low-level API like `copy_from_user` to interfere with the data being copied.

Reply:
I agree that it's an unusual place, but as I mentioned in the first post: We use user-provided `timespec`s as-is in many places, and I think trying to hunt them all down is a bad idea.

What would be a better design? I considered`Optional<timespec> copy_timespec_from_user(const timespec*)` for a while, but disregarded it for the above reason.

Reply:
Could you static assert that T is not timespec in the regular `copy_from_user`, with a message that people should use copy_timespec_from_user? 
Best of both worlds?

Reply:
Turns out, static asserts *always* trigger, even if the template isn't instantiated. Huh! So I went with ` __attribute__((deprecated("use " #REPLACEMENT " instead")))` instead.